### PR TITLE
fix(hudcomponents): don't wait four frames

### PR DESCRIPTION
**Pull request by NotSomething0**
Waiting even just one frame causes blinking which can be seen [here](https://i.imgur.com/KGXkhOX.mp4)

### CODE REVIEW

File: client/hudcomponents.lua [hunk 1]
```diff
@@ -1,30 +1,19 @@
...
-		DisplayAmmoThisFrame(true)
+-- 1 : WANTED_STARS | 4 : MP_CASH | 13 : CASH_CHANGE | 21 : HUD_COMPONENTS
+-- 2 : WEAPON_ICON | 7 : AREA_NAME | 14 : RETICLE | 22 : HUD_WEAPONS
+-- 3 : CASH | 9 : STREET_NAME | 17 : SAVING_GAME
 
-		Wait(4)
+local DISABLED_COMPONENTS <const> = {1, 2, 3, 4, 7, 9, 13, 14, 17, 21, 22}
+CreateThread(function()
+  while true do
+    for _, component in pairs(DISABLED_COMPONENTS) do
```
Comment (BerkieBb) [inline client/hudcomponents.lua:8]: Use iterator instead
```lua
for i = 1, #DISABLED_COMPONENTS do
    HideHudComponentThisFrame(DISABLED_COMPONENTS[i]
end
```
This is faster than using pairs

**Review by BerkieBb — CHANGES_REQUESTED** [review]

**Comment by BerkieBb** [comment]
Just so you know, Wait accepts milliseconds as an argument, not frames, frames are rather a sort of like part where code gets executed, the amount of time a frame takes relies on the performance of the system and the code it has to run, on higher end systems, frames can take less longer than lower end systems because they have more performance and can run everything faster

**Comment by NotSomething0** [comment]
Yeah usually they get used interchangeably when used in conversation thanks for the explanation though :)

**Comment by BerkieBb** [comment]
One more thing btw, the indentation is at 2 spaces, can you please change it to 4 spaces?

**Comment by NotSomething0** [comment]
Should be good now

**Comment by BerkieBb** [comment]
Added in #250

**Pull request closed without merging**